src/Reflection/ReflectionClass.php

```diff
@@ -242,7 +242,15 @@ public static function createFromNode(
      */
     public function getShortName() : string
     {
-        return $this->node->name;
+        if (!$this->isAnonymous()) {
+            return $this->node->name;
+        }
+
+        $name = 'class@anonymous';
+        if ($this->getFileName() !== null) {
```

Comment:
Note: this isn't due to performance, but due to the fact that the PHP type system has no immutability/idempotent guarantees